lib/membrane/video_compositor/wgpu/wgpu_adapter.ex

```diff
@@ -107,14 +109,36 @@ defmodule Membrane.VideoCompositor.Wgpu do
           state :: wgpu_state_t(),
           id :: id_t(),
           placement :: RustStructs.VideoPlacement.t()
-        ) :: :ok
+        ) :: :ok | {:error, :bad_video_index}
   def update_placement(state, id, placement) do
     case Native.update_placement(state, id, placement) do
       :ok -> :ok
+      {:error, {:bad_video_index, _index}} -> {:error, :bad_video_index}
```

Comment:
It's used to handle `update_placement` call before calling `handle_caps` for specific input pad. If pad would call `handle_pad_added` -> `update_placement` -> `handle_caps` we would raise instead of updating `initial_video_placements`.